spec/components/site_navbar_component_spec.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe SiteNavbarComponent, type: :component do
+  subject(:rendered) do
+    Capybara::Node::Simple.new(render_inline(described_class.new).to_s)
+  end
+
+  before do
+    # rubocop:disable RSpec/AnyInstance
+    allow_any_instance_of(ApplicationController).to receive(:current_user).and_return(user)
+    # rubocop:enable RSpec/AnyInstance
```

Comment:
Does this work so we don't have to disable `RSpec/AnyInstance`?
```suggestion
    allow(controller).to receive(:current_user).and_return(user)
```

Reply:
Nope. Causes an error in CI

Reply:
https://github.com/sul-dlss/exhibits/actions/runs/11146554130/job/30978782641

Reply:
Oh, I think with `view_component` v3 `controller` needs to be `vc_test_controller`

Reply:
undefined local variable or method `vc_test_controller' for #<RSpec::ExampleGroups::SiteNavbarComponent::WithANonAdminUser:0x00000001261b9910>

Reply:
I think if you rebase this with what's currently in main you'll pick up view_components v3 and `vc_test_controller` will work. 🤞